src/weather_model_graphs/create/archetype.py

```diff
@@ -18,20 +20,29 @@ def create_keisler_graph(xy_grid, grid_refinement_factor=3):
 
     Parameters
     ----------
-    xy_grid: np.ndarray
-        2D array of grid point positions.
-    grid_refinement_factor: float
-        Refinement factor between grid points and mesh
+    coords: np.ndarray
+        2D array of grid point positions, either in-projection Cartesian coordinates or lat-lons
+    mesh_node_distance: float
+        Distance (in x- and y-direction) between created mesh nodes,
+        in coordinate system of coords
+    projection: cartopy.crs.CRS or None
+        Projection instance used to transform given lat-lon coords to in-projection
+        Cartesian coordinates. If None the coords are assumed to already be Cartesian.
+    decode_mask: Iterator or None
+        Mask describing which grid positions should be decoded to (included in the m2g subgraph).
+        It should have the same length as the number of grid position coordinates given in `coords`.
+        The mask being set to True means that corresponding grid nodes should be included in g2m.
+        If `decode_mask=None` (default), all grid nodes are included.
```

Comment:
Surely decode_mask should be an Iterable rather than an Iterator, right? Also maybe just to be completely clear, I suggested a comment about the mask specifying what is included in the output graph.

```suggestion
    decode_mask: Iterable or None
        Mask describing which grid positions should be decoded to (included in the m2g subgraph), i.e. which positions should be output.
        It should have the same length as the number of grid position coordinates given in `coords`.
        The mask being set to True means that corresponding grid nodes should be included in g2m.
        If `decode_mask=None` (default), all grid nodes are included.
```

Reply:
Yes, good catch :smile: